compiler/src/iree/compiler/Dialect/VM/IR/VMOps.cpp

```diff
@@ -314,14 +314,15 @@ void ImportOp::build(OpBuilder &builder, OperationState &result, StringRef name,
                       builder.getStringAttr(name));
   result.addAttribute("function_type", TypeAttr::get(type));
   result.attributes.append(attrs.begin(), attrs.end());
-  if (argAttrs.empty()) {
-    return;
+  if (!argAttrs.empty()) {
+    assert(type.getNumInputs() == argAttrs.size() &&
+           "expected as many argument attribute lists as arguments");
+    function_interface_impl::addArgAndResultAttrs(builder, result, argAttrs,
+                                                  /*resultAttrs=*/llvm::None);
   }
 
-  assert(type.getNumInputs() == argAttrs.size() &&
-         "expected as many argument attribute lists as arguments");
-  function_interface_impl::addArgAndResultAttrs(builder, result, argAttrs,
-                                                /*resultAttrs=*/llvm::None);
+  // No clue why this is required.
```

Comment:
Because your ImportOp defines it: `let regions = (region AnyRegion:$body);`

And FunctionOpInterface requires a single region, differentiated between decl/def based on whether it has a block. From the definition of FunctionOpInterface:

```
      - must have a single region, that may be comprised with multiple blocks,
        that corresponds to the function body.
        * when this region is empty, the operation corresponds to an external
          function.
        * leading arguments of the first block of the region are treated as
          function arguments.
```

So maybe remove the comment or change it to "Always requires an empty region to comply with FunctionOpInterface external functions"

Reply:
(fixing in follow-on)